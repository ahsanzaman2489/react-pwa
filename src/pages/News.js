import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import * as NewsActions from '../actions/newsActions';
import * as qs from 'query-string';

// import {NO_DATA} from '../../constants/app';


export class NewsPage extends Component {
    state = {
        query: this.props.location.search
    };

    componentDidMount() {
        const {location, fetchNewsSources} = this.props;
        const query = qs.parse(location.search);
        fetchNewsSources('sources', '&category=' + query.category);
    }

    componentWillReceiveProps(nextProps) {
        const {location, fetchNewsSources} = nextProps;
        const query = qs.parse(location.search);


        if (this.state.query !== location.search) {
            fetchNewsSources('sources', '&category=' + query.category);
            this.setState({query: location.search})
        }
    }


    render() {
        return (

            <div className="car-detail">
                News
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    console.log(state)
    return {
        sources: state.headlineReducers,
    };
};

function mapDispatchToProps(dispatch) {
    return {...bindActionCreators(NewsActions, dispatch)}
}


export default connect(mapStateToProps, mapDispatchToProps)(NewsPage);
