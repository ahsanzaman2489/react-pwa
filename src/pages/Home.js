import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import * as NewsActions from '../actions/News';

// import {NO_DATA} from '../../constants/app';


class HomePage extends Component {
    state = {

    };

    componentDidMount() {

    }


    render() {

        return (

            <div className="car-detail">
                Home
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        car: state,
    };
};

function mapDispatchToProps(dispatch) {
    return {...bindActionCreators({NewsActions}, dispatch)}
}


export default connect(mapStateToProps, mapDispatchToProps)(HomePage);
