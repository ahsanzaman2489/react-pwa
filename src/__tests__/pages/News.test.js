import React from 'react';
import {shallow, mount} from 'enzyme/build';
import {NewsPage} from '../../pages/News';


const setup = (props = {}, state = null) => {
    const defaultProps = {
        location: {search: "category=health"},
        news: {data: {articles: []}},
        fetchNewsSources: jest.fn(),
        fetchNews: jest.fn(),
        sources: {
            data: [
                "argaam",
                "australian-financial-review",
                "bloomberg"
            ]
        }
    };
    const setUpProps = {...defaultProps, ...props};
    const wrapper = shallow(<NewsPage {...setUpProps}/>);
    const tree = () => setTimeout(() => renderer.create(<NewsPage {...setUpProps}/>).toJSON());
    if (state) wrapper.setState(state);
    return {wrapper, props: setUpProps, tree};
};

let wrapper, props;

beforeEach(() => {
    const setupWrapper = setup({}, {query: location.search, samePage: true});
    wrapper = setupWrapper.wrapper;
    props = setupWrapper.props;
});

afterEach(() => {
    wrapper.unmount();
});

describe("News component", () => {
    it('Should render', () => {
        expect(wrapper.find('.news-detail').length).toBe(1);
    });

    it('componentDidMount Should call new actions to fetch with category', () => {
        expect(props.fetchNewsSources).toHaveBeenCalled();
        expect(props.fetchNewsSources).toHaveBeenCalledTimes(1);
        expect(props.fetchNewsSources).toHaveBeenCalledWith('sources', props.location.search);
    });

    it('Equal array function should return true or false if equal or not', () => {
        const {arraysEqual} = wrapper.instance();
        expect(arraysEqual(['a', 'b'], ['a', 'b'])).toBeTruthy();
        expect(arraysEqual(['a', 'b'], ['a', 'c'])).toBeFalsy();
    });


    it('componentDidUpdate should update component if category and sources change', () => {
        const spyGetNewPerSource = jest.spyOn(wrapper.instance(), "getNewPerSource");
        const newData = {
            data: [
                "argaam",
                "australian-financial-review",
            ]
        };
        wrapper.setProps({
            sources: newData,
            location: {search: "category=game"}
        });

        expect(spyGetNewPerSource).toBeCalled();
        expect(spyGetNewPerSource).toBeCalledWith(newData.data, {search: "category=game"});
        //
        // expect(props.fetchNews).toBeCalled();
        // expect(props.fetchNews).toBeCalledWith('everything', 'sources=' + [
        //     "argaam",
        //     "australian-financial-review",
        // ]);
    });

});

