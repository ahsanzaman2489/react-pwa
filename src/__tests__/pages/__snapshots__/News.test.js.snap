// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News component should match with snapshot 1`] = `7`;
